### ISSUE TITLE
FFI plugins should be able to access system dlls in sandbox

### ISSUE THREAD

**Issue by dcharkes** [flutter/flutter#114676]
`flutter create --template plugin_ffi --platform=windows` + `flutter build windows --release` requires manually copying system dlls.

This is rather undesirable.

We should investigate how to allow opening system dlls from the sandbox.

> > The flutter create --template=ffi_plugin --platform=windows does work in the sandbox?
>
> So I created a plugin using this command, `cd` to the example folder, ran `flutter build windows --release`, and then copied the `Release` folder that contains the executable into the sandbox, and see for yourself that it complains first about system32 dlls:
![image](https://user-images.githubusercontent.com/49842031/199929880-002ede67-a984-4e4b-a34c-670a1c922b4d.png)
> 
> Now I will copy the dlls and run the app again:
>
> ![image](https://user-images.githubusercontent.com/49842031/199931025-c91e6178-037c-4dd5-b5af-a15d08455e1c.png)

_Originally posted by @Haidar0096 in https://github.com/flutter/flutter/issues/114641#issuecomment-1303132940_

**Comment by dcharkes** [comment]
I don't believe this an issue with Dart. This is an issue with the Windows sandbox requiring _all_ dependencies to be bundled . This might be an issue with normal Flutter apps which are not FFI plugins as well. We probably have never tried flutter apps in this Windows sandbox.

**Comment by Haidar0096** [comment]
> I don't believe this an issue with Dart. This is an issue with the Windows sandbox requiring _all_ dependencies to be bundled . This might be an issue with normal Flutter apps which are not FFI plugins as well. We probably have never tried flutter apps in this Windows sandbox.

So given that the release build ran on my machine, and given what yoy said, I can be sure that distributing the release folder (for example) will be enough for the .exe to work?

**Comment by stuartmorgan** [comment]
> This is an issue with the Windows sandbox requiring _all_ dependencies to be bundled . This might be an issue with normal Flutter apps which are not FFI plugins as well.

I would imagine that there's a difference between linked dependencies and calling `DynamicLibrary.open` on a path at runtime. It's implausible that all system libraries have to replicated in every sandboxed application.

**Comment by xkeyC** [comment]
https://learn.microsoft.com/en-US/cpp/windows/latest-supported-vc-redist?view=msvc-170

By default, Microsoft Visual C++ Redistributable (msvcpxxx.dll) is not shipped with Windows. The sandbox environment is a clean new system, so this runtime is not included.

The sandbox does not cause this dll to fail to load, but it does not exist on a clean Windows system. This is still true on a brand new system outside the sandbox.

**Comment by loic-sharma** [comment]
@dcharkes If I'm understanding the issue correctly, it sounds like you didn't copy the C++ redistributables in your app. This is necessary before deploying your app to all Windows machines, some of which might not have C++ runtime installed. See our instructions here: https://docs.flutter.dev/platform-integration/windows/building#building-your-own-zip-file-for-windows

I use the Windows sandbox all the time to test Flutter, so this is definitely a supported scenario :)

Please let me know if I misunderstood the issue!

**Comment by dcharkes** [comment]
@loic-sharma With the work on https://github.com/flutter/flutter/issues/129757 being in preview, I think we should tell users to use `template=package_ffi`. We're aiming to deprecate and remove `plugin_ffi` once build hooks are in stable (https://github.com/flutter/flutter/issues/131209).

**Comment by loic-sharma** [comment]
@dcharkes That sounds good! I'll close this off for now since I believe this is not a bug (see [this comment](https://github.com/flutter/flutter/issues/114676#issuecomment-3134297253)), but please feel free to reopen this if necessary! :)
